.github/workflows/test-os.yml

```diff
@@ -52,7 +52,7 @@ jobs:
         name: Test
         env:
           TMPDIR: ${{ runner.temp }}
-          SKIP_INTEGRATION_TESTS: 1
+          #SKIP_INTEGRATION_TESTS: 1
```

Comment:
There seem to be missing requirements in the test env on Windows. Since we're enabling integration tests, we need to make sure we have the registry binary installed before we kick off the tests, along with any missing binaries that are needed.

This can be done as an extra step in the test-os job.

https://github.com/moby/buildkit/actions/runs/7177401305/job/19543810673?pr=4479#step:5:2331

Reply:
Sure, `registry.exe` was missing, I've added a step for that.

Reply:
Ok, few more binaries needed, will add those: `containerd.exe`, `buildkitd.exe`;

Mmh, wondering, when it comes to `buildkitd.exe`, should I checkout that from the remote branch (PR's branch) since there could be changes made there?

Reply:
For buildkitd.exe you must build the code that's already being checked out here:

https://github.com/moby/buildkit/blob/master/.github/workflows/test-os.yml#L40

The CWD should already be set to the checked out code. You can use a `run` step to simply:

```bash
go build -o c:/Windows/System32/buildkitd.exe ./cmd/buildkitd.exe
```

(You'll probably want a different path)

Here is an example of using the `run` action:

https://github.com/containerd/containerd/blob/main/.github/workflows/windows-periodic.yml#L60

For containerd.exe, you can parse `go.mod` to get the required containerd version and wget the released binary from the containerd releases page, here:

https://github.com/containerd/containerd/releases

Buildkit generally uses stable releases of containerd. Alternatively, you can checkout the containerd tag for the version in `go.mod` and build from source.

You will also need to make sure they are placed somewhere in `$PATH`.

Reply:
$env:GOPATH is available with the [Setup Go](https://github.com/moby/buildkit/blob/f84cfe3d6ab7b0645df76e82a02458d3088f32a9/.github/workflows/test-os.yml#L43) step, so we're good with `go install`.

Reply:
Sure, I will explore the `go.mod` parsing idea, seems elegant. For now, I'd just put the version manually through `env`.

Reply:
Done:
```powershell
$goModFileContent = Get-Content go.mod -Raw
$pattern = '(?<=github\.com/containerd/containerd v)[\d\.]+'
$version = [regex]::Match($goModFileContent, $pattern).Value
```